lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -1159,11 +1133,21 @@ static void optimizeArithmeticNodes(Function *F) {
     Node *node = worklist.back();
     worklist.pop_back();
 
-    auto *sn = simplifyNode(node, F);
-    if (sn != node) {
-      node->getNthResult(0).replaceAllUsesOfWith(sn);
+    auto *SN = simplifyNode(node, F);
+    if (SN != node) {
+      node->getNthResult(0).replaceAllUsesOfWith(SN);
```

Comment:
Maybe as a follow-up PR, could we feature a Node::replaceAllUsesOfWith that would replace each result of the node with the related results of the other nodes. Hardcoding the number of results makes me nervous.

We don't have arithmetic nodes with more than one result yet, but if we do at some point (e.g., something equivalent to LLVM UDIVREM that returns both the quotient and remainder), this is going to be wrong.

For now, we should assert that we only have one result. (We could leave it this way until the assert breaks too.)